config/v1/types_infrastructure.go

```diff
@@ -229,6 +229,17 @@ const (
 	IBMCloudProviderTypeUPI IBMCloudProviderType = "UPI"
 )
 
+// ClusterHostedDNS indicates whether the cluster DNS is hosted by the cluster or Core DNS .
+type ClusterHostedDNS string
+
+const (
+	// ClusterHostedDNSEnabled indicates that the DNS solution is hosted by the cluster.
+	ClusterHostedDNSEnabled ClusterHostedDNS = "Enabled"
+
+	// ClusterHostedDNSDisabled indicates that the DNS solution is hosted by Core DNS.
```

Comment:
ClusterHostedDNSDisabled indicates that the cluster is using the DNS solution provided by the cloud platform.